File: reference/TutorialCode/dotNet/CSharpExamples/Code/configuration/ConfigurationExamples.cs
using Db4objects.Db4o;
using Db4objects.Db4o.Config;

namespace Db4oTutorialCode.Code.Configuration
{
    public class ConfigurationExamples
    {
        public void ConfigureDb4O()
        {
            // #example: Important configuration switches
            IEmbeddedConfiguration configuration = Db4oEmbedded.NewConfiguration();
            // #end example


            // #example: A few examples
            // Add index
            configuration.Common.ObjectClass(typeof (Driver)).Indexed(true);
            // #end example

            // #example: Finally pass the configuration container factory
            IObjectContainer container = Db4oEmbedded.OpenFile(configuration, "database.db4o");
            // #end example
        }

        private class Driver
        {
        }
    }
}

File: reference/TutorialCode/dotNet/VisualBasicExamples/Code/configuration/ConfigurationExamples.vb
Imports Db4objects.Db4o
Imports Db4objects.Db4o.Config

Namespace Db4oTutorialCode.Code.Configuration
    Public Class ConfigurationExamples
        Public Sub ConfigureDb4O()
            ' #example: Important configuration switches
            Dim configuration As IEmbeddedConfiguration = Db4oEmbedded.NewConfiguration()
            ' #end example


            ' #example: A few examples
            ' Add index
            configuration.Common.ObjectClass(GetType(Driver)).Indexed(True)
            ' #end example

            ' #example: Finally pass the configuration container factory
            Dim container As IObjectContainer = Db4oEmbedded.OpenFile(configuration, "database.db4o")
            ' #end example
        End Sub

        Private Class Driver
        End Class
    End Class
End Namespace
